### DISCUSSION TITLE
No inspection available with m2o while creation

### DISCUSSION THREAD

**Discussion by anna-osipchyk** [aminalaee/sqladmin#441] [Q&A]
Hello! I have a problem with no inspection error. I have a model User that has role_id as fk to Role model. When i create new user and all data is valid (form is valid), sqladmin just generates new instance and returns RedirectResponsem that's great. But when my form is not valid, then create.html is generated. but it need's to inspect this fk, and as far as it's just "str", not Role instance, it trhows this error:
sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <class 'str'>.

I tried sending requests with AsyncClient  
    role = await t.AsyncRoleFactory(name="user") //just instanse of Role model
data={
                "username": "test@email.com",
                "created_at": (datetime.datetime.now() - datetime.timedelta(days=14)).strftime("%Y-%m-%d %H:%M:%S"),
                "role": role,
            },
and it worked great. Please, share your thoughts, thank you!

**Comment by aminalaee** [comment]
Hello @anna-osipchyk 

Can you share a very minimal example of your model?
And also explain that you are using the Admin UI right? Or testing with AsyncClient?

**Reply by anna-osipchyk** [reply]
Thank you for your response! Yes, sure, take a look.

class User(Base):
    __tablename__ = "user"

    id = Column(Integer, primary_key=True)
    email = Column(EmailType, nullable=True, unique=True, index=True) #i use form validator for this field
    phone_number = Column(String(15), nullable=True, unique=True, index=True)#and for this as well
    password_hash = Column(String(128), nullable=False)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.datetime.now)
    role = relationship("Role", back_populates="users")
    role_id = Column(Integer, ForeignKey("role.id", ondelete="SET NULL"), nullable=True)

class Role(Base):
    __tablename__ = "role"

    id = Column(Integer, primary_key=True)
    name = Column(String(50), nullable=False, unique=True, index=True)
    permissions = relationship("Permission", secondary=role_permission, back_populates="roles")
    users = relationship("User", back_populates="role")


I use admin UI, that's true. but i need to test my code, that's why i use AsyncClient just for testing. But of course the main usage of admin panel is via UI

**Reply by aminalaee** [reply]
@anna-osipchyk Can you please share what is the version of sqladmin you are using?
I can't reproduce the same issue.

**Reply by anna-osipchyk** [reply]
Sure! sqladmin version == 0.8.0
I have been using this library for 3 months
